doc/design/katib_on_sqlflow.md

```diff
@@ -0,0 +1,60 @@
+# Katib on SQLFlow
+This design is about optimizing model hyperparameters in SQLFlow by creating Katib step via couler.
+
+## Requirements:
+
+Currently, SQLFlow allows programmers to use SQL queries to enable ML model training, prediction, and explanation. However, hyperparameters optimization is also a necessary function that can help users to tune hyperparamters of their models in an easy way. This function will be supported by Katib in SQLFlow.
+
+## Hyperparameter Optimization in Katib
+
+[Katib](https://github.com/kubeflow/katib) is a Kubernetes Native System for Hyperparameter Tuning and Neural Architecture Search. The system is inspired by Google Vizier and supports multiple ML/DL frameworks (e.g. TensorFlow, Apache MXNet, and PyTorch) and models (e.g., XGBoost).
```

Comment:
Need to add a para to tell what a Katib job, the inputs, the outputs, and what the job can do.